File: android/app/src/main/java/me/bryanoltman/listentowikipedia/model/WikipediaLanguage.kt
package me.bryanoltman.listentowikipedia.model

data class WikipediaLanguage(val name: String, val code: String)

object WikipediaLanguages {
    val arabic = WikipediaLanguage("Arabic", "ar")
    val assamese = WikipediaLanguage("Assamese", "as")
    val belarusian = WikipediaLanguage("Belarusian", "be")
    val bengali = WikipediaLanguage("Bengali", "bn")
    val bulgarian = WikipediaLanguage("Bulgarian", "bg")
    val chinese = WikipediaLanguage("Chinese", "zh")
    val dutch = WikipediaLanguage("Dutch", "nl")
    val english = WikipediaLanguage("English", "en")
    val farsi = WikipediaLanguage("Farsi", "fa")
    val french = WikipediaLanguage("French", "fr")
    val german = WikipediaLanguage("German", "de")
    val gujarati = WikipediaLanguage("Gujarati", "gu")
    val hebrew = WikipediaLanguage("Hebrew", "he")
    val hindi = WikipediaLanguage("Hindi", "hi")
    val indonesian = WikipediaLanguage("Indonesian", "id")
    val italian = WikipediaLanguage("Italian", "it")
    val japanese = WikipediaLanguage("Japanese", "ja")
    val kannada = WikipediaLanguage("Kannada", "kn")
    val macedonian = WikipediaLanguage("Macedonian", "mk")
    val malayalam = WikipediaLanguage("Malayalam", "ml")
    val marathi = WikipediaLanguage("Marathi", "mr")
    val oriya = WikipediaLanguage("Oriya", "or")
    val polish = WikipediaLanguage("Polish", "pl")
    val punjabi = WikipediaLanguage("Punjabi", "pa")
    val russian = WikipediaLanguage("Russian", "ru")
    val sanskrit = WikipediaLanguage("Sanskrit", "sa")
    val serbian = WikipediaLanguage("Serbian", "sr")
    val spanish = WikipediaLanguage("Spanish", "es")
    val swedish = WikipediaLanguage("Swedish", "sv")
    val tamil = WikipediaLanguage("Tamil", "ta")
    val telugu = WikipediaLanguage("Telugu", "te")
    val ukrainian = WikipediaLanguage("Ukrainian", "uk")

    val all: List<WikipediaLanguage> = listOf(
        arabic,
        assamese,
        belarusian,
        bengali,
        bulgarian,
        chinese,
        dutch,
        english,
        farsi,
        french,
        german,
        gujarati,
        hebrew,
        hindi,
        indonesian,
        italian,
        japanese,
        kannada,
        macedonian,
        malayalam,
        marathi,
        oriya,
        polish,
        punjabi,
        russian,
        sanskrit,
        serbian,
        spanish,
        swedish,
        tamil,
        telugu,
        ukrainian
    )
}

File: ListenToWikipedia/Models/WikipediaLanguage.swift
import Foundation

struct WikipediaLanguage: Identifiable, Hashable, Comparable {
  let name: String
  let code: String

  var id: String { code }

  static func < (lhs: WikipediaLanguage, rhs: WikipediaLanguage) -> Bool {
    lhs.name < rhs.name
  }

  static let all: [WikipediaLanguage] = [
    WikipediaLanguage(name: "Arabic", code: "ar"),
    WikipediaLanguage(name: "Assamese", code: "as"),
    WikipediaLanguage(name: "Belarusian", code: "be"),
    WikipediaLanguage(name: "Bengali", code: "bn"),
    WikipediaLanguage(name: "Bulgarian", code: "bg"),
    WikipediaLanguage(name: "Chinese", code: "zh"),
    WikipediaLanguage(name: "Dutch", code: "nl"),
    WikipediaLanguage(name: "English", code: "en"),
    WikipediaLanguage(name: "Farsi", code: "fa"),
    WikipediaLanguage(name: "French", code: "fr"),
    WikipediaLanguage(name: "German", code: "de"),
    WikipediaLanguage(name: "Gujarati", code: "gu"),
    WikipediaLanguage(name: "Hebrew", code: "he"),
    WikipediaLanguage(name: "Hindi", code: "hi"),
    WikipediaLanguage(name: "Indonesian", code: "id"),
    WikipediaLanguage(name: "Italian", code: "it"),
    WikipediaLanguage(name: "Japanese", code: "ja"),
    WikipediaLanguage(name: "Kannada", code: "kn"),
    WikipediaLanguage(name: "Macedonian", code: "mk"),
    WikipediaLanguage(name: "Malayalam", code: "ml"),
    WikipediaLanguage(name: "Oriya", code: "or"),
    WikipediaLanguage(name: "Polish", code: "pl"),
    WikipediaLanguage(name: "Punjabi", code: "pa"),
    WikipediaLanguage(name: "Russian", code: "ru"),
    WikipediaLanguage(name: "Sanskrit", code: "sa"),
    WikipediaLanguage(name: "Serbian", code: "sr"),
    WikipediaLanguage(name: "Spanish", code: "es"),
    WikipediaLanguage(name: "Swedish", code: "sv"),
    WikipediaLanguage(name: "Tamil", code: "ta"),
    WikipediaLanguage(name: "Telugu", code: "te"),
    WikipediaLanguage(name: "Ukrainian", code: "uk"),
    WikipediaLanguage(name: "Marathi", code: "mr"),
  ]
}
